src/buffer/out/textBuffer.cpp

```diff
@@ -2353,3 +2354,86 @@ void TextBuffer::CopyHyperlinkMaps(const TextBuffer& other)
     _hyperlinkMap = other._hyperlinkMap;
     _hyperlinkCustomIdMap = other._hyperlinkCustomIdMap;
 }
+
+// Method Description:
+// - Adds a regex pattern we should search for
+// - The searching does not happen here, we only search when asked to by TerminalCore
+// Arguments:
+// - The regex pattern
+// Return value:
+// - An ID that the caller should associate with the given pattern
+const size_t TextBuffer::AddPatternRecognizer(const std::string regexString)

```

Comment:
This made a copy on the way in because the signature is `std::string` instead of `std::string_view`, and it's going to make another copy during the make_pair/emplace step. You might want to accept a `const std::string_view` to get rid of some of the copies.